src/components/input-fields/simple-input/SimpleInput.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { InputHTMLAttributes } from "react";
+import { SErrorMessage } from "../styles/SErrorMessage";
+import { SLabel } from "../styles/SLabel";
+import { SSimpleInput } from "./styles/SSimpleInput";
+import { SSimpleInputRequiredIndicator } from "./styles/SSimpleInputRequiredIndicator";
+import { SSimpleInputRequiredIndicatorContainer } from "./styles/SSimpleInputRequiredIndicatorContainer";
+import { SSimpleInputTextWidth } from "./styles/SSimpleInputTextWidth";
+import { SSimpleInputWrapper } from "./styles/SSimpleInputWrapper";
+
+type TSimpleInputState = "success" | "error";
+
+interface ISimpleInput extends InputHTMLAttributes<HTMLInputElement> {
+  label?: string;
+  errors?: string[];
+  state?: TSimpleInputState;
+  translationBadge?: JSX.Element;
+  required?: boolean;
+  onChange: any;
+  value: any;
+}
+
+export const SimpleInput: React.FC<ISimpleInput> = ({ state, required, children, value, onChange, ...props }) => {
+  const [active, setActive] = React.useState(false);
+  const [inputLength, setInputLength] = React.useState(value?.toString().length);
+
+  const inputRef = React.useRef<any>();
+  const spanRef = React.useRef<any>();
+
+  const onClick = () => {
```

Comment:
For all these handler methods use the hook useCallback, passing the dependencies.

onClick
resizeInput
handleFocus
handleBlur